src/Integration.Vsix/CFamily/CLangAnalyzer.cs

```diff
@@ -58,28 +63,40 @@ public bool IsAnalysisSupported(IEnumerable<AnalysisLanguage> languages)
         }
 
         public void ExecuteAnalysis(string path, string charset, IEnumerable<AnalysisLanguage> detectedLanguages,
-            IIssueConsumer consumer, ProjectItem projectItem, IAnalyzerOptions analyzerOptions,
+            IIssueConsumer consumer, IAnalyzerOptions analyzerOptions,
             CancellationToken cancellationToken)
         {
+            var projectItem = dte?.Solution?.FindProjectItem(path);
+            if (projectItem == null)
+            {
+                return;
+            }
```

Comment:
This is all that's required to fetch the `ProjectItem`, and this is the only place it's required.

The VS threading analyzers complain that this object should only be accessed on the UI thread. When manually testing I discovered that this is not always the case.
However, this isn't a new issue: this class was already accessing the object on the UI thread in the previous version of the code. I've created an issue: #1476.

Reply:
Is it possible that it also somehow caused the goldbar?

Reply:
It might, but I think it's more likely to cause a crash/deadlock.